src/Cassandra/Requests/RequestHandler.cs

```diff
@@ -112,9 +112,44 @@ private static IEnumerable<HostShard> GetQueryPlan(ISession session, IStatement
             // Single host iteration
             var host = (statement as Statement)?.Host;
 
-            return host == null
-                ? lbp.NewQueryPlan(session.Keyspace, statement)
-                : Enumerable.Repeat(new HostShard(host, -1), 1);
+            if (host != null)
+            {
+                return Enumerable.Repeat(new HostShard(host, -1), 1);
+            }
+
+            if ((statement as Statement)?.IsLwt() == true)
+            {
+                return GetReplicas(session.Keyspace, statement, lbp.NewQueryPlan(session.Keyspace, statement), session.Cluster);
+            }
+
+            return lbp.NewQueryPlan(session.Keyspace, statement);
+        }
```

Comment:
Let's move this logic to LoadBalancing policies, let's move it out to a separate PR.